x/mongo/driver/topology/connection_test.go

```diff
@@ -115,21 +115,65 @@ func TestConnection(t *testing.T) {
 				assert.NotNil(t, err, "expected connect error %v, got nil", want)
 				assert.Equal(t, want, got, "expected error %v, got %v", want, got)
 			})
-			t.Run("cancelConnectContext is nil after connect", func(t *testing.T) {
```

Comment:
The `context.CancelFunc` gets unpinned by both `connection.connect()` and `connection.closeConnectContext`, so I removed this test and split it into two separate ones to test both of these cases.